File: android/src/main/java/ai/nextbillion/maps_flutter/NextbillionMapOptionsSink.java

package ai.nextbillion.maps_flutter;


import ai.nextbillion.maps.geometry.LatLngBounds;

/** Receiver of NBMap configuration options. */
interface NextbillionMapOptionsSink {
  void setCameraTargetBounds(
      LatLngBounds bounds); // todo: dddd replace with CameraPosition.Builder target

  void setCompassEnabled(boolean compassEnabled);

  // TODO: styleString is not actually a part of options. consider moving
  void setStyleString(String styleString);

  void setMinMaxZoomPreference(Float min, Float max);

  void setRotateGesturesEnabled(boolean rotateGesturesEnabled);

  void setScrollGesturesEnabled(boolean scrollGesturesEnabled);

  void setTiltGesturesEnabled(boolean tiltGesturesEnabled);

  void setTrackCameraPosition(boolean trackCameraPosition);

  void setZoomGesturesEnabled(boolean zoomGesturesEnabled);

  void setMyLocationEnabled(boolean myLocationEnabled);

  void setMyLocationTrackingMode(int myLocationTrackingMode);

  void setMyLocationRenderMode(int myLocationRenderMode);

  void setLogoViewMargins(int x, int y);

  void setCompassGravity(int gravity);

  void setCompassViewMargins(int x, int y);

  void setAttributionButtonGravity(int gravity);

  void setAttributionButtonMargins(int x, int y);
}

File: ios/Classes/NextbillionMapOptionsSink.swift

import Nbmap

protocol NextbillionMapOptionsSink {
    func setCameraTargetBounds(bounds: NGLCoordinateBounds?)
    func setCompassEnabled(compassEnabled: Bool)
    func setStyleString(styleString: String)
    func setMinMaxZoomPreference(min: Double, max: Double)
    func setRotateGesturesEnabled(rotateGesturesEnabled: Bool)
    func setScrollGesturesEnabled(scrollGesturesEnabled: Bool)
    func setTiltGesturesEnabled(tiltGesturesEnabled: Bool)
    func setTrackCameraPosition(trackCameraPosition: Bool)
    func setZoomGesturesEnabled(zoomGesturesEnabled: Bool)
    func setMyLocationEnabled(myLocationEnabled: Bool)
    func setMyLocationTrackingMode(myLocationTrackingMode: NGLUserTrackingMode)
    func setMyLocationRenderMode(myLocationRenderMode: MyLocationRenderMode)
    func setLogoViewMargins(x: Double, y: Double)
    func setCompassViewPosition(position: NGLOrnamentPosition)
    func setCompassViewMargins(x: Double, y: Double)
    func setAttributionButtonMargins(x: Double, y: Double)
    func setAttributionButtonPosition(position: NGLOrnamentPosition)
}
